### PR TITLE
Add content API endpoints

### PR DESCRIPTION
**Pull request by idonov8**
Adding new endpoints

### CODE REVIEW

File: components/schemas/File.yaml [hunk 1]
```diff
@@ -2,19 +2,27 @@ type: object
...
+      - file
+      - dir
+    example: 
+      - file
   size:
     type: integer
     format: int64
     example: 421
-  hash:
+  sha:
```
Comment (simonlsk) [inline components/schemas/File.yaml]: I think it's hash where does the sha come from?

File: spec.yaml [hunk 2]
```diff
@@ -2,16 +2,93 @@
...
+                    },
+                  ]
+        '404':
+          description: Non existing path
+    put:
+      tags:
+        - Content
+      summary: Upload data to a repository
+      description: |
+        **`last_commit`** - If the tip of the branch differs on the server at the moment of processing the request, the request is denied.
```
Comment (simonlsk) [inline spec.yaml]: Actually if the upload contains only new files, then `last_commit` is not mandatory and won't be "denied".

**Review by simonlsk — CHANGES_REQUESTED** [review]

File: components/schemas/File.yaml (hunk 1 above)
Comment (idonov8) [reply components/schemas/File.yaml]: The response I got when testing had `sha`

Comment (simonlsk) [reply components/schemas/File.yaml]: You're right

Comment (simonlsk) [reply components/schemas/File.yaml]: We need to fix that

Comment (simonlsk) [reply components/schemas/File.yaml]: PR open - https://github.com/DagsHub/gogs/pull/1007

**Review by simonlsk — APPROVED** [review]

**Pull request merged**